nocturne/envs/base_env.py

```diff
@@ -149,6 +158,15 @@ def step(  # pylint: disable=arguments-renamed,too-many-locals,too-many-branches
             veh_id = veh_obj.getID()
             if veh_id in self.done_ids:
                 continue
+
+            # Remove vehicle from the scene if position is invalid (but not collided or goal achieved?)
```

Comment:
I'm guessing that this condition, which is post reset, is only getting triggered for vehicles that are expert controlled. The expert controlled vehicles will occasionally pop to the invalid position